ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1019,154 @@ def test_server_option_with_unreachable_ad(self):
+        # Related https://bugzilla.redhat.com/show_bug.cgi?id=2010512
+        tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_nonposixuser_nondefault_primary_group(self, type):
+        """
+        Test for non default primary group
+        """
+        nonposixuser1 = "nonposixuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser1)
+        if type == "true":
+            assert uid == gid
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_posix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust-posix'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_coresponding_existing_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) but there is no group with the
+        corresponding gidNumber.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.master, posixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_only_uid_number_auto_private_group_default(self, type):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, auto-private-group
+        is set to false/true/hybrid
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        if type == "true":
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert uid == gid
+        else:
+            for host in [self.master, self.clients[0]]:
+                result = host.run_command(['id', posixuser], raiseonerr=False)
+                tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_group_primary_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) and there is primary group
+        with gid number defined..
+        """
+        posixuser = "testuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        if type == "true":
+            assert uid == gid
```

Comment:
You can also check that uid=10042 (in the 3 cases) and gid = 10042 (true) or 10047 (hybrid/false)